File: backend/src/main/java/com/examplatform/classroom/Classroom.java
package com.examplatform.classroom;

import com.examplatform.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "classrooms")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Classroom {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private User teacher;

    @ManyToMany
    @JoinTable(
            name = "classroom_students",
            joinColumns = @JoinColumn(name = "classroom_id"),
            inverseJoinColumns = @JoinColumn(name = "student_id")
    )
    @Builder.Default
    private Set<User> students = new HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/Classroom.ts
export interface Classroom {
  id: number;
  name: string;
  code: string;
  description?: string;
  teacherId: number;
  teacherName: string;
  studentIds: number[];
  studentNames: string[];
  studentCount: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateClassroomDto {
  name: string;
  code: string;
  description?: string;
  teacherId?: number;
  studentIds?: number[];
}

export interface UpdateClassroomDto {
  name: string;
  code: string;
  description?: string;
  teacherId?: number;
  studentIds?: number[];
}
